### ISSUE TITLE
Backport Finalizer on apps for cluster-monitoring and monitoring operator not getting cleaned up #19529

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#19900]
Backport for https://github.com/rancher/rancher/issues/19529

See also https://github.com/rancher/rancher/issues/19899

**Comment by alena1108** [comment]
Can be tested on rc5

**Comment by jiaqiluo** [comment]
The bug is reproduced on Rancher: v2.2.2 observing that the system project's namespace is stuck in terminating state in Rancher's kubernete
```
# kubectl get ns
NAME                 STATUS        AGE
cattle-global-data   Active        35m
cattle-system        Active        35m
default              Active        35m
kube-public          Active        35m
kube-system          Active        35m
p-gq4df              Terminating   32m
user-7wr5k           Active        34m
```
The bug fix is validated on v2.2.3-rc5 observing that the system project's namespace is deleted successfully
